### PR TITLE
Tests - scaffold views_settings suite

### PR DESCRIPTION
**Pull request by lucatume**
Tickets: 

* https://moderntribe.atlassian.net/browse/TEC-3208
* https://moderntribe.atlassian.net/browse/TEC-3209

Based on: https://github.com/moderntribe/tribe-testing-facilities/pull/21

This PR scaffolds an integrations suite dedicated to Views v2 and settings.

I've opted for a different suite to keep the code distinct from others and speed up debug.

This PR scope **is not** to scaffold all the tests we require, just to kickstart them.

### CODE REVIEW

File: src/Tribe/Views/V2/Views/Month_View.php [hunk 1]
```diff
@@ -373,7 +373,7 @@ static function ( $event ) use ( $date_object ) {
 	/**
 	 * {@inheritDoc}
 	 */
-	protected function calculate_grid_start_end( $date ) {
+	public function calculate_grid_start_end( $date ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Month_View.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Public method name "Month_View::calculate_grid_start_end" is not in camel caps format
* :no_entry_sign: Opening brace should be on a new line

<sub>hash: 8f106bb783befbb55c96d09a9fc9cfaa</sub>

**Comment by lucatume** [comment]
Refreshed from `release/B20.02`.

**Comment by lucatume** [comment]
Latest modifications rely on [`moderntribe/tribe-testing-facilities` PR](https://github.com/moderntribe/tribe-testing-facilities/pull/23)

File: tests/views_settings/_bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+use Tribe\Events\Views\V2\Service_Provider;
+
+// Let's  make sure Views v2 are activated if not.
+putenv( 'TRIBE_EVENTS_V2_VIEWS=1' );
+tribe_register_provider( Service_Provider::class );
+
+// Let's make sure to set rewrite rules.
+global $wp_rewrite;
+$wp_rewrite->permalink_structure = '/%postname%/';
```
Comment (bordoni) [inline tests/views_settings/_bootstrap.php]: Should we save this on the DB option too?

File: src/Tribe/Repositories/Event.php [hunk 3]
```diff
@@ -1183,11 +1191,8 @@ protected function update_date_meta( array $postarr, $post_id = null ) {
 			);
 
 			// Let's make sure the timezone is always a string.
-			if ( isset( $postarr['meta_input']['_EventTimezone'] ) ) {
-				$postarr['meta_input']['_EventTimezone'] = Timezones::build_timezone_object(
-					$postarr['meta_input']['_EventTimezone']
-				)->getName();
-			}
+			$postarr['meta_input']['_EventTimezone'] = Timezones::build_timezone_object( $input_timezone )->getName();
```
Comment (bordoni) [inline src/Tribe/Repositories/Event.php]: Wouldn't this create a problem of always saving a timezone? Which is a diff that than current behavior where we only modify if the timezone field is modified?

File: tests/views_settings/_bootstrap.php (hunk 2 above)
Comment (lucatume) [reply tests/views_settings/_bootstrap.php]: I think not, it currently works as intended.

File: src/Tribe/Repositories/Event.php (hunk 3 above)
Comment (lucatume) [reply src/Tribe/Repositories/Event.php]: Under the hood, we use the `update_meta_data` function that, for single custom fields like the `_EventTimezone` one, will not write the value to db if it's the same.

This example test will pass:
```php
public function should_not_write_the_same_value_twice() {
	$event         = static::factory()->event->create( [ 'meta_input' => [ '_EventTimezone' => 'America/New_York' ] ] );
	$updated_times = 0;

	add_action( 'update_postmeta', static function ( $meta_id, $object_id, $meta_key ) use ( &$updated_times ) {
		if ( $meta_key !== '_EventTimezone' ) {
			return null;
		}

		if ( ++ $updated_times > 1 ) {
			throw new AssertionFailedError( 'I should not be called twice' );
		}
	}, 10, 3 );

	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );
	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );
	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );

	$this->assertEquals( 1, $updated_times );
}
```

**Pull request merged**